### PR TITLE
NGAlert: Send resolve signal to alertmanager on alerting -> Normal

### PR DESCRIPTION
**Pull request by kylebrandt**
**What this PR does / why we need it**:
Sends a Postable Alert to to the embeded alert manager when Grafana managed alert instance 

**Which issue(s) this PR fixes**:

Not sure it entirely fixes everything with notifications that have been reported, but think at least this is a big part of it:

https://github.com/grafana/grafana/issues/36644 
https://github.com/grafana/grafana/issues/36888
https://github.com/grafana/grafana/issues/36037

**Special notes for your reviewer**:
Will replace https://github.com/grafana/grafana/pull/37311 I think, which includes this as well as removing resend delay, whereas this keeps resend delay.

### CODE REVIEW

File: pkg/services/ngalert/schedule/compat.go [hunk 1]
```diff
@@ -28,34 +28,35 @@ func FromAlertStateToPostableAlerts(logger log.Logger, firingStates []*state.Sta
...
-				StartsAt:    strfmt.DateTime(alertState.StartsAt),
-				EndsAt:      strfmt.DateTime(alertState.EndsAt),
-				Alert: models.Alert{
-					Labels:       models.LabelSet(nL),
-					GeneratorURL: strfmt.URI(genURL),
-				},
-			})
-			alertState.LastSentAt = ts
-			sentAlerts = append(sentAlerts, alertState)
+		if !alertState.NeedsSending(stateManager.ResendDelay) {
```
Comment (kylebrandt) [inline pkg/services/ngalert/schedule/compat.go:31]: This is a style change, continue to dedent code.

File: pkg/services/ngalert/state/manager.go [hunk 2]
```diff
@@ -175,6 +175,14 @@ func (st *Manager) setNextState(alertRule *ngModels.AlertRule, result eval.Resul
 	case eval.Pending: // we do not emit results with this state
 	}
 
+	// Set Resolved property so the scheduler knows to send a postable alert
+	// to alert manager.
```
Comment (gotjosh) [inline pkg/services/ngalert/state/manager.go:179]:
```suggestion
	// to Alertmanager.
```

File: pkg/services/ngalert/state/manager.go [hunk 3]
```diff
@@ -175,6 +175,14 @@ func (st *Manager) setNextState(alertRule *ngModels.AlertRule, result eval.Resul
...
+	if oldState == eval.Alerting && currentState.State == eval.Normal {
```
Comment (kylebrandt) [inline pkg/services/ngalert/state/manager.go:180]: can just be `currentState.Resolved = oldState == eval.Alerting && currentState.State == eval.Normal` I think

Comment (gotjosh) [inline pkg/services/ngalert/state/manager.go:180]: Apologies as I'm not really familiar with all of the states, but I'd assume this should be:

```
oldState == eval.Alerting && currentState.State != eval.Alerting
```

Unless there's no chance we can go from alerting -> pending? (e.g. in the case of a restart)

File: pkg/services/ngalert/state/state.go [hunk 4]
```diff
@@ -112,10 +113,13 @@ func (a *State) resultNoData(alertRule *ngModels.AlertRule, result eval.Result)
 }
 
 func (a *State) NeedsSending(resendDelay time.Duration) bool {
-	if a.State != eval.Alerting {
```
Comment (gotjosh) [inline pkg/services/ngalert/state/state.go:115]: I think we have some good unit tests for NeedsSending, could we add a case or two to make sure this works as expected? 😄

**Review by gotjosh — COMMENTED** [review]
LGTM, just have a few nits - nice work @kylebrandt.

File: pkg/services/ngalert/state/manager.go (hunk 3 above)
Comment (kylebrandt) [reply pkg/services/ngalert/state/manager.go:180]: I think we don't sent notifications on Error / or NoData states (@davidmparrott ?) So I don't transitions to those states would be recoveries.

Comment (davidmparrott) [reply pkg/services/ngalert/state/manager.go:180]: Correct, notifications are not sent for NoData or Error states. Users can configure the NoData state to resolve to Alerting if they want to get notified, though.

**Review by davidmparrott — APPROVED** [review]
LGTM thanks for getting this fix in so quickly.

**Pull request merged**